apps/src/music/player/MusicPlayer.js

```diff
@@ -187,39 +187,39 @@ export default class MusicPlayer {
     return this.soundEvents;
   }
 
+  /*
   getCurrentAudioElapsedTime() {
     if (!this.isPlaying) {
       return 0;
     }
 
     return GetCurrentAudioTime() - this.startPlayingAudioTime;
   }
+  */
 
-  getPlayheadPosition() {
-    if (!this.isPlaying) {
-      return 0;
-    }
-
-    // Playhead time is 1-based (user-facing)
-    return 1 + this.getCurrentAudioElapsedTime() / this.secondsPerMeasure();
-  }
-
-  getCurrentMeasure() {
+  // Called by generated code in the interpreter, and returns the current
+  // playhead, in floating point for an exact position, and scaled to measures.
+  // It's 1-based.
+  // Returns 0 if music is not playing.
+  getCurrentPlayhead() {
     const currentAudioTime = GetCurrentAudioTime();
     if (!this.isPlaying || currentAudioTime === null) {
-      return -1;
+      return 0;
     }
 
-    return this.convertSecondsToMeasure(
-      currentAudioTime - this.startPlayingAudioTime
+    return (
+      1 +
+      this.convertSecondsToMeasure(
+        currentAudioTime - this.startPlayingAudioTime
+      )
     );
   }
 
   playSoundEvent(soundEvent, onStop) {
     if (soundEvent.type === EventType.PLAY) {
       const eventStart =
         this.startPlayingAudioTime +
-        this.convertMeasureToSeconds(soundEvent.when);
+        this.convertMeasureToSeconds(soundEvent.when - 1);
```

Comment:
nit: can you add a comment for why this adjustment is needed?

Reply:
Moved the adjustment into `convertPlayheadPositionToSeconds`, mentioned below.